### PR TITLE
ulog_parser: fixed parsing of array topics

### PR DESCRIPTION
**Pull request by RomanBapst**
**Problem**
I was not able to open log files which included arrays of nested topics. I think I found the cause why this was not working, however, the data of the nested topics is still not displayed correctly. There seems to be a miss-alignment between labels and data (only for that nested topic).

Unfortunately, I'm currently not able to share a log file that includes nested topics. (The one I was using was from a customer and I cannot share it here :-( ).
I will try to look for a public log. In my case the vehicle was logging the "esc_status" message which contains an array of "esc_report" message.

This is the error message that I saw when trying to parse the log file:
image

With this PR the log file is parsed and the message label appear, however, the data is not correct yet.
In the plot above I wanted to plot temperature but in fact the graph corresponds to the esc current.
image

@facontidavide Maybe you can help me get this right.

Signed-off-by: Roman <bapstroman@gmail.com>

### CODE REVIEW

**Comment by facontidavide** [comment]
Thank you for reporting this issue. I do not have time to look at this right now, but I hope I will be able to fix it by the end of this month

**Comment by RomanBapst** [comment]
@facontidavide I just found the problem and pushed a fix. The handling of the padding fields was not done properly. The esc_status messages has a padding fields before the nested message and that is what was causing the problem.
This works properly now.

File: plugins/DataLoadULog/ulog_parser.cpp [hunk 1]
```diff
@@ -736,10 +748,15 @@ ULogParser::Timeseries ULogParser::createTimeseries(const ULogParser::Format* fo
     {
         for( const auto& field: format.fields)
         {
+            // skip padding messages
+            if (field.field_name == "_padding0") {
```
Comment (bkueng) [inline plugins/DataLoadULog/ulog_parser.cpp]: You should skip all fields starting with `_padding`, see https://dev.px4.io/en/log/ulog_file_format.html#definitions-section

Comment (RomanBapst) [reply plugins/DataLoadULog/ulog_parser.cpp]: @bkueng Thanks, will do!

Comment (RomanBapst) [reply plugins/DataLoadULog/ulog_parser.cpp]: Done

**Comment by facontidavide** [comment]
Thanks a lot. Awesome work.

File: plugins/DataLoadULog/ulog_parser.cpp (hunk 1 above)
Comment (bkueng) [reply plugins/DataLoadULog/ulog_parser.cpp]: I guess you forgot that one?

Comment (facontidavide) [reply plugins/DataLoadULog/ulog_parser.cpp]: I took care of it

Comment (bkueng) [reply plugins/DataLoadULog/ulog_parser.cpp]: Thank you.

**Pull request merged**